tests/postgres-types-time.rs

```diff
@@ -0,0 +1,108 @@
+use sqlx::types::time::{OffsetDateTime, Date, Time, UtcOffset};
+use sqlx::{Connection, PgConnection, Row};
+
+async fn connect() -> anyhow::Result<PgConnection> {
+    Ok(PgConnection::open(dotenv::var("DATABASE_URL")?).await?)
+}
+
+#[cfg_attr(feature = "runtime-async-std", async_std::test)]
+#[cfg_attr(feature = "runtime-tokio", tokio::test)]
+async fn postgres_timers_date() -> anyhow::Result<()> {
+    let mut conn = connect().await?;
+
+    let value = Date::try_from_ymd(2019, 1, 2).unwrap();
+
+    let row = sqlx::query("SELECT DATE '2019-01-02' = $1, $1")
+        .bind(&value)
+        .fetch_one(&mut conn)
+        .await?;
+
+    assert!(row.get::<bool, _>(0));
+    assert_eq!(value, row.get(1));
+
+    Ok(())
+}
+
+#[cfg_attr(feature = "runtime-async-std", async_std::test)]
+#[cfg_attr(feature = "runtime-tokio", tokio::test)]
+async fn mysql_timers_date_time() -> anyhow::Result<()> {
+    let mut conn = connect().await?;
+
+    let value = Date::try_from_ymd(2019, 1, 2)
+        .unwrap()
+        .try_with_hms(5, 10, 20)
+        .unwrap();
+
+    let row = sqlx::query("SELECT '2019-01-02 05:10:20' = $1, $1")
+        .bind(&value)
+        .fetch_one(&mut conn)
+        .await?;
+
+    assert!(row.get::<bool, _>(0));
+    assert_eq!(value, row.get(1));
+
+    Ok(())
+}
+
+#[cfg_attr(feature = "runtime-async-std", async_std::test)]
+#[cfg_attr(feature = "runtime-tokio", tokio::test)]
+async fn postgres_timers_time() -> anyhow::Result<()> {
+    let mut conn = connect().await?;
+
+    let value = Time::try_from_hms_micro(5, 10, 20, 115100).unwrap();
+
+    let row = sqlx::query("SELECT TIME '05:10:20.115100' = $1, TIME '05:10:20.115100'")
+        .bind(&value)
+        .fetch_one(&mut conn)
+        .await?;
+
+    // FIXME: line 60 fails, while assertion on line 61 holds true (???)
+    // assert!(row.get::<bool, _>(0));
+    assert_eq!(value, row.get(1));
```

Comment:
I'm yet to determine the cause of this behaviour

Reply:
Postgres should print queries to the stderr of the server daemon by default, you could try attaching to that and see what Postgres is interpreting the value as.

Reply:
Thanks for the suggestion!
There was an error in Time objects encoding, combined with lack of unit tests, covering Time/NaiveTime encoding/decoding. Fixed it + added UTs